xr_sample_project/Assets/SampleViewer/Samples/XRTableTop/Scripts/XRTableTopInteractor.cs

```diff
@@ -0,0 +1,304 @@
+using Esri.ArcGISMapsSDK.Components;
+using Esri.ArcGISMapsSDK.Samples.Components;
+using Esri.HPFramework;
+using Unity.Mathematics;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.XR;
+using UnityEngine.XR.Interaction.Toolkit;
+using CommonUsages = UnityEngine.XR.CommonUsages;
+using InputDevice = UnityEngine.XR.InputDevice;
+
+public class XRTableTopInteractor : MonoBehaviour
+{
+    [SerializeField] private Animation anim;
+    [SerializeField] private ArcGISMapComponent arcGISMapComponent;
+    private Camera camera;
+    private Vector3 dragStartPoint = Vector3.zero;
+    private double4x4 dragStartWorldMatrix;
+    private bool handTrackingEnabled = false;
+    [SerializeField] private HPRoot hpRoot;
+    private bool isDragging = false;
+    private bool rightHanded = true;
+    [SerializeField] private float radiusScalar = 50f;
+    [SerializeField] private ArcGISTabletopControllerComponent tableTop;
+    [SerializeField] private GameObject tableTopWrapper;
+
+    [Header("Material")]
+    [SerializeField] private Color color;
+    [SerializeField] private Material mat;
+    [SerializeField] private bool useTexture;
+
+    [Header("Hand Input")]
+    [SerializeField] private XRNode leftInputSource;
+    [SerializeField] private XRRayInteractor leftControllerInteractor;
+    [SerializeField] private XRRayInteractor leftHandInteractor;
+    private Vector2 leftInputAxis;
+    private RaycastHit leftHandHit;
+    private RaycastHit leftControllerHit;
+    [SerializeField] private XRNode rightInputSource;
+    [SerializeField] private XRRayInteractor rightControllerInteractor;
+    [SerializeField] private XRRayInteractor rightHandInteractor;
+    private Vector2 rightInputAxis;
+    private RaycastHit rightHandHit;
+    private RaycastHit rightControllerHit;
+    [SerializeField] private InputActionProperty selectLeft;
+    [SerializeField] private InputActionProperty selectRight;
+
+    private void Awake()
+    {
+        camera = GetComponentInChildren<Camera>();
+        mat.SetColor("_Color", color);
+        if (useTexture)
+        {
+            mat.SetInt("_UseTexture", 1);
+        }
+        else
+        {
+            mat.SetInt("_UseTexture", 0);
+        }
+    }
+
+    public void EndPointDrag()
+    {
+        isDragging = false;
+    }
+
+    public void PlayAnimation()
+    {
+        anim.Play();
+    }
+
+    private void Start()
+    {
+#if UNITY_EDITOR
+        camera.clearFlags = CameraClearFlags.Skybox;
+#elif UNITY_STANDALONE_WIN
+        camera.clearFlags = CameraClearFlags.Skybox;
+#else
+        camera.clearFlags = CameraClearFlags.Color;
+#endif
+    }
+
+    public void StartDragPoint()
+    {
+        if (!handTrackingEnabled)
+        {
+            StartPointDragController();
+        }
+        else
+        {
+            StartPointDragHand();
+        }
+    }
+
+    public void StartPointDragController()
+    {
+        if (rightHanded)
```

Comment:
There is a ton of repeat code in this file that can be avoided by selecting the left or right hand interactor, then executing the code.
`var controllerInteractor = rightHanded ? rightControllerInteractor : leftControllerInteractor;`